capabilityreceiver/pkg/capabilityhandler/validation/base.go

```diff
@@ -170,28 +189,203 @@ func (bv *baseVal) KindValid(capStat capabilityparser.CapabilityStatement) []end
+	}
+	// if document is not nil, add to actual list
+	document, err := capStat.GetDocument()
+	if err == nil && len(document) > 0 {
+		actualVal = append(actualVal, "document")
+	}
+	// If none of the above exist, the capability statement is not valid
+	if len(actualVal) == 0 {
+		ruleError.Actual = ""
+		ruleError.Valid = false
+		return ruleError
+	}
+	ruleError.Actual = strings.Join(actualVal, ",")
 	return ruleError
 }
 
+// DescribeEndpointValid checks the requirement: "A Conformance Resource/Capability Statement SHALL have at least one of description,
+// software, or implementation element."
 func (bv *baseVal) DescribeEndpointValid(capStat capabilityparser.CapabilityStatement) endpointmanager.Rule {
-	var ruleError endpointmanager.Rule
+	var actualVal []string
+	baseComment := "A Conformance Resource SHALL have at least one of description, software, or implementation element."
+	ruleError := endpointmanager.Rule{
+		RuleName:  endpointmanager.DescribeEndptRule,
+		Valid:     true,
+		Expected:  "description,software,implementation",
+		Comment:   baseComment,
+		Reference: "http://hl7.org/fhir/DSTU2/conformance.html",
+		ImplGuide: "USCore 3.1",
+	}
+
+	if capStat == nil {
+		ruleError.Valid = false
+		ruleError.Comment = "The Capability Statement does not exist; cannot check description, software, or implementation elements."
+		return ruleError
+	}
+
+	// If description is not an empty string, add to actual list
+	description, err := capStat.GetDescription()
+	if err == nil && len(description) > 0 {
+		actualVal = append(actualVal, "description")
+	}
+	// If software is not nil, add to actual list
+	software, err := capStat.GetSoftware()
+	if err == nil && len(software) > 0 {
+		actualVal = append(actualVal, "software")
+	}
+	// if implementation is not nil, add to actual list
+	implementation, err := capStat.GetImplementation()
+	if err == nil && len(implementation) > 0 {
+		actualVal = append(actualVal, "implementation")
+	}
+	// If none of the above exist, the capability statement is not valid
+	if len(actualVal) == 0 {
+		ruleError.Actual = ""
+		ruleError.Valid = false
+		return ruleError
+	}
+	ruleError.Actual = strings.Join(actualVal, ",")
 	return ruleError
 }
 
+// DocumentSetValid checks the requirement: "The set of documents must be unique by the combination of profile and mode."
 func (bv *baseVal) DocumentSetValid(capStat capabilityparser.CapabilityStatement) endpointmanager.Rule {
-	var ruleError endpointmanager.Rule
+	baseComment := "The set of documents must be unique by the combination of profile and mode."
+	ruleError := endpointmanager.Rule{
+		RuleName:  endpointmanager.DocumentValidRule,
+		Valid:     false,
+		Expected:  "true",
+		Actual:    "false",
+		Comment:   baseComment,
+		Reference: "http://hl7.org/fhir/DSTU2/conformance.html",
+		ImplGuide: "USCore 3.1",
+	}
+
+	if capStat == nil {
+		ruleError.Comment = "The Capability Statement does not exist; cannot check documents."
```

Comment:
I think in this check and elsewhere, in this file, it should say "Conformance Resource" instead of "Capability Statement"